src/output-json-alert.c

```diff
@@ -217,6 +217,25 @@ void AlertJsonHeader(const Packet *p, const PacketAlert *pa, json_t *js)
     json_object_set_new(js, "alert", ajs);
 }
 
+static void AlertJsonParent(const Packet *p, json_t *js)
+{
+    json_t *parent = json_object();
+    if (parent == NULL)
+        return;
+
+    if (p->root == NULL)
+        return;
+
+    /* get a lock to access root packet fields */
+    SCMutex *m = &p->root->tunnel_mutex;
+
+    SCMutexLock(m);
+    JsonFiveTuple((const Packet *)p->root, 0, parent);
```

Comment:
could be interesting to add in Packet::recursion_level. If we have multiple tunnel layers it will show how deep this packet is.